### PR TITLE
Minimum stake schedule

### PR DESCRIPTION
**Pull request by pdyraga**
Closes: keep-network/keep-ecdsa#302

See: https://github.com/keep-network/keep-core/pull/1499
See: https://github.com/keep-network/sortition-pools/pull/74
See: https://github.com/keep-network/sortition-pools/pull/78

Minimum stake is no longer a system-wide constant and it changes over time in `TokenStaking`. `BondedECDSAKeepFactory` needs to be updated to support the minimum stake schedule.

The factory does no longer have the minimum stake hardcoded. To check if the operator has a minimum stake, it queries `TokenStaking` contract.

Also, the sortition pool now expects the recent value of the minimum stake to be passed for each group selection. This is happening in `openKeep` function.

Factory declares the eligible stake divisor for sortition pool to be `1 KEEP`.

### CODE REVIEW

File: solidity/test/BondedECDSAKeepFactoryTest.js [hunk 1]
```diff
@@ -307,44 +309,39 @@ contract("BondedECDSAKeepFactory", async (accounts) => {
...
+      await stakeOperators(members, minimumStake.subn(1))
 
-      assert.isTrue(
+      assert.isFalse(
         await keepFactory.isOperatorUpToDate(members[0], application)
       )
     })
 
-    it("returns false if the operator stake is above minimum", async () => {
+    it("returns false if the operator stake increased", async () => {
```
Comment (nkuba) [inline solidity/test/BondedECDSAKeepFactoryTest.js:337]:
```suggestion
    it("returns true if the operator stake increased", async () => {
```
?

File: solidity/test/BondedECDSAKeepFactoryTest.js [hunk 2]
```diff
@@ -307,44 +309,39 @@ contract("BondedECDSAKeepFactory", async (accounts) => {
...
         from: members[0],
       })
 
-      // We multiply minimumStake as sortition pools expect multiplies of the
-      // minimum stake to calculate stakers weight for eligibility.
-      await stakeOperators(members, minimumStake.mul(new BN(2)))
+      await stakeOperators(members, minimumStake.addn(1))
 
-      assert.isFalse(
+      assert.isTrue(
```
Comment (nkuba) [inline solidity/test/BondedECDSAKeepFactoryTest.js:344]: Could you please explain this change? Have some rules changed on the sortition pool side?

File: solidity/test/BondedECDSAKeepFactoryTest.js (hunk 1 above)
Comment (pdyraga) [reply solidity/test/BondedECDSAKeepFactoryTest.js]: https://github.com/keep-network/keep-ecdsa/pull/377/commits/7b311d767ff61f09f1853fec5dcfb012230efab7

File: solidity/test/BondedECDSAKeepFactoryTest.js (hunk 2 above)
Comment (pdyraga) [reply solidity/test/BondedECDSAKeepFactoryTest.js]: I rethought tests after your comment and pushed more precise cases in https://github.com/keep-network/keep-ecdsa/pull/377/commits/3a61bf47809ab3b8fa01c27fc79473fa80934464

From https://github.com/keep-network/sortition-pools/pull/74:

> Minimum stake is no longer a hardcoded value and can change over time. To accommodate potential changes in minimum stake, we now use a constant pool weight divisor for bonded and standard sortition pool just like in the fully backed pool.
> 
> The initial minimum stake value is passed as a pool constructor parameter and the current value is later passed as an argument to select group function.

We agreed with @eth-r to set the pool weight divisor to 1 KEEP. There is an additional factor here: EVM performs implicit floor for `uint256` division. As a result:
- if stake drops by `10^-(18) KEEP`, `isOperatorUpToDate` returns `false`; this is covered by `returns false if the operator stake dropped` case
- if stake drops by `1 KEEP`, `isOperatorUpToDate` returns `false`; this is not covered by a separate case but is obvious from the previous one
- if stake increases by `1 KEEP`, `isOperatorUpToDate` returns `false`; this is covered by `returns false if the operator stake increased` case
- if stake increases by `10^(-18) KEEP`, `isOperatorUpToDate` returns `true`; this is covered by `returns true if the operator stake increase is below weight precision` case

File: solidity/test/BondedECDSAKeepFactoryTest.js (hunk 1 above)
Comment (pdyraga) [reply solidity/test/BondedECDSAKeepFactoryTest.js]: Nope, reworked in https://github.com/keep-network/keep-ecdsa/pull/377/commits/3a61bf47809ab3b8fa01c27fc79473fa80934464

**Pull request merged**